### PR TITLE
add mul.out, add.out, sub.out, div.out

### PR DESCRIPTION
**Pull request by taozhiwei**
add mul.out,add.out,sub.out,div.out

### CODE REVIEW

File: src/flag_gems/ops/add.py [hunk 1]
```diff
@@ -93,6 +93,29 @@ def add(A, B, *, alpha=1):
         return torch.tensor(A + B * alpha)
 
 
+def add_out(A, B, *, alpha=1, out):
+    logger.debug("GEMS ADD_OUT")
+    if not (isinstance(A, torch.Tensor) and isinstance(B, torch.Tensor)):
+        raise TypeError("aten::add.out expects tensor operands")
```
Comment (tengqm) [inline src/flag_gems/ops/add.py:99]: I'm not confident that this is correct.
See `add_` implementation.
We are dealing with tensor and scalar mixtures in that path.

**Review by tengqm — COMMENTED** [review]
@taozhiwei Please consider splitting this PR into four PRs, each PR with implementation, accuracy tests and performance benchmarks.
